File: src/main/java/gg/mineral/practice/scoreboard/impl/TeamBoxingScoreboard.java
package gg.mineral.practice.scoreboard.impl;

import gg.mineral.practice.entity.Profile;
import gg.mineral.practice.match.TeamMatch;
import gg.mineral.practice.scoreboard.Scoreboard;
import gg.mineral.practice.scoreboard.ScoreboardHandler;
import gg.mineral.practice.util.messages.CC;

public class TeamBoxingScoreboard
        implements Scoreboard {
    public static final Scoreboard INSTANCE = new TeamBoxingScoreboard();

    @Override
    public void updateBoard(ScoreboardHandler board, Profile profile) {
        board.updateTitle(CC.PRIMARY + CC.B + "Mineral");
        if (profile.getMatch() == null) {
            return;
        }

        TeamMatch match = (TeamMatch) profile.getMatch();

        boolean isTeam1 = match.getTeam1RemainingPlayers().contains(profile);
        int hitCount = isTeam1 ? match.getTeam1HitCount() : match.getTeam2HitCount();
        int opponentHitCount = isTeam1 ? match.getTeam2HitCount() : match.getTeam1HitCount();
        int requiredHitCount = isTeam1 ? match.getTeam1RequiredHitCount() : match.getTeam2RequiredHitCount();
        int opponentRequiredHitCount = isTeam1 ? match.getTeam2RequiredHitCount() : match.getTeam1RequiredHitCount();

        board.updateLines(CC.BOARD_SEPARATOR,
                CC.ACCENT + "Your Hits: " + CC.SECONDARY + hitCount + "/" + requiredHitCount,
                CC.ACCENT + "Their Hits: " + CC.SECONDARY + opponentHitCount + "/" + opponentRequiredHitCount,
                CC.SPACER,
                CC.SECONDARY + "mineral.gg",
                CC.BOARD_SEPARATOR);
    }
}

File: src/main/kotlin/gg/mineral/practice/scoreboard/impl/TeamBoxingScoreboard.kt
package gg.mineral.practice.scoreboard.impl

import gg.mineral.practice.entity.Profile
import gg.mineral.practice.match.TeamMatch
import gg.mineral.practice.match.appender.MatchAppender
import gg.mineral.practice.scoreboard.Scoreboard
import gg.mineral.practice.scoreboard.ScoreboardHandler
import gg.mineral.practice.util.messages.CC

class TeamBoxingScoreboard : Scoreboard, MatchAppender {
    override fun updateBoard(board: ScoreboardHandler, profile: Profile) {
        board.updateTitle(CC.PRIMARY + CC.B + "Mineral")

        profile.match?.let {
            if (it is TeamMatch) {
                val isTeam1: Boolean = it.team1Players.all().contains(profile)
                val hitCount: Int = if (isTeam1) it.team1HitCount else it.team2HitCount
                val opponentHitCount: Int = if (isTeam1) it.team2HitCount else it.team1HitCount
                val requiredHitCount: Int = if (isTeam1) it.team1RequiredHitCount else it.team2RequiredHitCount
                val opponentRequiredHitCount: Int = if (isTeam1)
                    it.team2RequiredHitCount
                else
                    it.team1RequiredHitCount

                board.updateLines(
                    CC.BOARD_SEPARATOR,
                    CC.ACCENT + "Your Hits: " + CC.SECONDARY + hitCount + "/" + requiredHitCount,
                    CC.ACCENT + "Their Hits: " + CC.SECONDARY + opponentHitCount + "/" + opponentRequiredHitCount,
                    CC.SPACER,
                    CC.SECONDARY + "mineral.gg",
                    CC.BOARD_SEPARATOR
                )
            }
        }
    }

    companion object {
        val INSTANCE: Scoreboard = TeamBoxingScoreboard()
    }
}
